core/src/test/java/org/mskcc/cbio/portal/util/TestRequestBodyGZipFilter.java

```diff
@@ -0,0 +1,174 @@
+package org.mskcc.cbio.portal.util;
+
+import com.google.common.net.HttpHeaders;
+import org.junit.Assert;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mockito;
+import org.springframework.test.util.ReflectionTestUtils;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletInputStream;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.stream.Collectors;
+
+public class TestRequestBodyGZipFilter {
+
+    /**
+     * When I filter a request that doesn't have the gzip header,
+     * the request should not be wrapped, and chain.doFilter should
+     * be called on the same request object passed to the method.
+     */
+    @Test
+    public void testDoFilterEnabledNotGZipped() throws Exception {
+        RequestBodyGZipFilter subject = createRequestBodyGZipFilter(true, 50000000);
+
+        HttpServletRequest request = createHttpServletRequest(null, "POST");
```

Comment:
The accepted strings for httpMethod can be referenced symbolically here perhaps? (from quick web search):
https://javaee.github.io/javaee-spec/javadocs/javax/ws/rs/HttpMethod.html